src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -0,0 +1,115 @@
+from datetime import datetime
+from typing import Callable, Dict, Mapping, Optional
+
+from arroyo import Topic
+from arroyo.backends.kafka import KafkaConsumer, KafkaPayload
+from arroyo.processing import StreamProcessor
+from arroyo.processing.strategies import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.types import Message, Partition, Position
+from django.conf import settings
+
+from sentry.constants import DataCategory
+from sentry.sentry_metrics.indexer.strings import TRANSACTION_METRICS_NAMES
+from sentry.utils import json
+from sentry.utils.kafka_config import get_kafka_consumer_cluster_options
+from sentry.utils.outcomes import Outcome, track_outcome
+
+
+def get_metrics_billing_consumer(
+    topic: str,
+    group_id: str,
+    auto_offset_reset: str,
+    **options,
+) -> StreamProcessor[KafkaPayload]:
+    # TODO: support force_topic and force_cluster
+    cluster_name = settings.KAFKA_TOPICS[topic]["cluster"]
+
+    return StreamProcessor(
+        consumer=KafkaConsumer(
+            get_kafka_consumer_cluster_options(
+                cluster_name=cluster_name,
+                override_params={
+                    "enable.auto.commit": False,
+                    "enable.auto.offset.store": False,
+                    "auto.offset.reset": auto_offset_reset,
+                    "group.id": group_id,
+                },
```

Comment:
Are there any other configuration parameters I should consider?

Reply:
Mind using this function to build the config? https://github.com/getsentry/arroyo/blob/main/arroyo/backends/kafka/configuration.py#L69 

Arroyo auto offset reset is a little bit different to normal and it allows us to also add the --strict-offset-reset later to our consumers.